packages/beacon-node/src/chain/archiveStore/observers/frequentStateArchiveObserver.ts

```diff
@@ -15,6 +15,7 @@ export class FrequentStateArchiveObserver extends QueueObserver {
   }
 
   async onForkChoiceFinalized(finalized: CheckpointWithHex): Promise<void> {
+    // should be after archiveBlocksObserver to handle restart cleanly
```

Comment:
Please put this line in archiveStore as well above where the `FrequentStateArchiveObserver` is registered on line 68